src/interactive-window/interactiveWindow.ts

```diff
@@ -95,7 +95,7 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
         return this._submitters;
     }
     public get notebookDocument(): NotebookDocument {
-        return this.notebookEditor.notebook;
+        return this.notebookEditor?.notebook;
```

Comment:
Would this change the typing to `notebookDocument: NotebookDocument | undefined`?

Reply:
It actually was already that, but I suppose I should go through with the rest of the corrections for that

Reply:
I think I'd rather tackle this in debt week - there is a lot of code in `InteractiveWindow` that relies on notebookDocument not being null, which would add a lot of extra `undefined` checking. Now that I've poked around quite a bit this iteration, I have a better idea of what could be done, including not relying on the editor to be the same throughout the lifetime of the IW object (separate issue).